### ISSUE TITLE
Suggest `Option::is_some` for `while let Some(_) = iterator.next()`

### ISSUE THREAD

**Issue by Luro02** [rust-lang/rust-clippy#5462]
```rust
#![warn(clippy::all, clippy::pedantic, clippy::nursery)]

fn main() {
    // should cause a warning that suggests replacing this with
    // while "".chars().next().is_some() {}
    while let Some(_) = "".chars().next() {}

    // causes a warning that suggests replacing this with
    // if Some("").is_some() {}
    if let Some(_) = Some("") {}
}
```

[playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=%23!%5Bwarn(clippy%3A%3Aall%2C%20clippy%3A%3Apedantic%2C%20clippy%3A%3Anursery)%5D%0A%0Afn%20main()%20%7B%0A%20%20%20%20%2F%2F%20should%20cause%20a%20warning%20that%20suggests%20replacing%20this%20with%0A%20%20%20%20%2F%2F%20while%20%22%22.chars().next().is_some()%20%7B%7D%0A%20%20%20%20while%20let%20Some(_)%20%3D%20%22%22.chars().next()%20%7B%7D%0A%0A%20%20%20%20%2F%2F%20causes%20a%20warning%20that%20suggests%20replacing%20this%20with%0A%20%20%20%20%2F%2F%20if%20Some(%22%22).is_some()%20%7B%7D%0A%20%20%20%20if%20let%20Some(_)%20%3D%20Some(%22%22)%20%7B%7D%0A%7D)

**Comment by alex-700** [comment]
Hi! Can I try to fix this? 
If I understand correctly it's just a forgotten case of `MatchSource` in [redundant_pattern_matching](https://rust-lang.github.io/rust-clippy/master/index.html#redundant_pattern_matching) lint. If it's ok to handle this similar to `MatchSource::IfLetDesugar`, I can do it.

**Comment by Luro02** [comment]
> Hi! Can I try to fix this?

Feel free to fix this :)

**Comment by flip1995** [comment]
Fixed in #5483
